.github/workflows/benchmarks.yml

```diff
@@ -0,0 +1,273 @@
+          path: benches/
+
+      - name: Combine benchmark results
+        run: |
+          # Create combined results file
+          echo "# Foundry Benchmark Results" > benches/LATEST.md
+          echo "" >> benches/LATEST.md
+          echo "Generated on: $(date -u)" >> benches/LATEST.md
+          echo "" >> benches/LATEST.md
+
+          # Add forge_test results if available
+          if [ -f "benches/forge-test-results/forge_test_results.md" ]; then
+            echo "## Forge Test and Fuzz Test Results" >> benches/LATEST.md
+            echo "" >> benches/LATEST.md
+            tail -n +2 benches/forge-test-results/forge_test_results.md >> benches/LATEST.md
+            echo "" >> benches/LATEST.md
+          fi
+
+          # Add forge_build results if available
+          if [ -f "benches/forge-build-results/forge_build_results.md" ]; then
+            echo "## Forge Build Results" >> benches/LATEST.md
+            echo "" >> benches/LATEST.md
+            tail -n +2 benches/forge-build-results/forge_build_results.md >> benches/LATEST.md
+            echo "" >> benches/LATEST.md
+          fi
+
+          # Add forge_coverage results if available
+          if [ -f "benches/forge-coverage-results/forge_coverage_results.md" ]; then
+            echo "## Forge Coverage Results" >> benches/LATEST.md
+            echo "" >> benches/LATEST.md
+            tail -n +2 benches/forge-coverage-results/forge_coverage_results.md >> benches/LATEST.md
+            echo "" >> benches/LATEST.md
+          fi
+
+          # Clean up artifact directories
+          rm -rf benches/forge-test-results
+          rm -rf benches/forge-build-results
+          rm -rf benches/forge-coverage-results
+
+      - name: Commit benchmark results
+        run: |
+          git config --local user.email "action@github.com"
+          git config --local user.name "GitHub Action"
+          git add benches/LATEST.md benches/forge_test_results.md benches/forge_build_results.md benches/forge_coverage_results.md 2>/dev/null || true
+          if git diff --staged --quiet; then
+            echo "No changes to commit"
+          else
+            git commit -m "chore(\`benches\`): update benchmark results
+
+            🤖 Generated with [Foundry Benchmarks](https://github.com/${{ github.repository }}/actions)
+            
+            Co-Authored-By: github-actions <github-actions@github.com>"
+            git push
+          fi
+
+      - name: Read benchmark results
+        id: benchmark_results
+        run: |
+          if [ -f "benches/LATEST.md" ]; then
+            {
+              echo 'results<<EOF'
+              cat benches/LATEST.md
+              echo 'EOF'
+            } >> $GITHUB_OUTPUT
+          else
+            echo 'results=No benchmark results found.' >> $GITHUB_OUTPUT
+          fi
+
+      - name: Comment on PR
+        if: github.event.inputs.pr_number != ''
+        uses: actions/github-script@v7
+        with:
+          script: |
```

Comment:
can we move as much logic as possible away from the yaml file into separate files